lib/open_food_network/scope_variants_for_search.rb

```diff
@@ -34,7 +34,9 @@ def search_params
     def query_scope
       Spree::Variant.where(is_master: false).
         includes(option_values: :option_type).
-        ransack(search_params.merge(m: 'or')).result
+        ransack(search_params.merge(m: 'or')).
+        result.
+        order("display_name")
```

Comment:
Is the `display_name` always filled? Should we use this field or the `display_as` field?

Reply:
It depends on what is used. Did you test this? Maybe add a spec!

I'm just guessing, but I found this form referenced the the add-product form:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/views/spree/admin/variants/_autocomplete.js.erb#L12

And the serializer defines the name like this:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/serializers/api/admin/variant_serializer.rb#L11-L17

But neither `name` nor `full_name` is not a column in the variants table. `name` comes from the product:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/models/spree/variant.rb#L22

And `full_name` is obscurely defined here:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/services/variant_units/variant_and_line_item_naming.rb#L49-L61

So the search order would be name, display_name, unit_to_display. This last attribute is computed as well:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/services/variant_units/variant_and_line_item_naming.rb#L69-L77

We can expand that to display_as, variant_unit_name, options_text. And now we reach a point that is probably not worth implementing the order for:
https://github.com/openfoodfoundation/openfoodnetwork/blob/6e905bd2fc21767017fc8c56792ed43d2e812810/app/services/variant_units/variant_and_line_item_naming.rb#L12-L26

I haven't tested this but kind of pseudo code could look like this:

```suggestion
        includes(:product).
        order("name, display_name, display_as, variant_unit_name")
```

Still not perfect but maybe it was never implemented because the perfect solution is too hard.

Reply:
Thanks a lot for you deep search, very appreciate... 🙏 

> Did you test this? Maybe add a spec!

Yep sure, and works fine for me, in my specific case, not sure this is the "normal" case ;)

I follow your advice: a5292d179